lib/guest_installation_and_configuration_base.pm

```diff
@@ -1254,8 +1254,8 @@ sub config_guest_installation_automation {
         $self->{guest_installation_automation_options} = "--extra-args \"inst.ks=$self->{guest_installation_automation_file}\"";
         $self->{guest_installation_automation_options} = "--extra-args \"ks=$self->{guest_installation_automation_file}\"" if (($self->{guest_os_name} =~ /oraclelinux/im) and ($self->{guest_version_major} lt 7));
     }
-    if (script_run("curl -sSf $self->{guest_installation_automation_file} > /dev/null") ne 0) {
-        record_info("Guest $self->{guest_name} unattended installation file hosted on local host can not be reached", "Mark guest installation as FAILED. The unattended installation file url is $self->{guest_installation_automation_file}");
+    if (script_retry("curl -sSf $self->{guest_installation_automation_file} > /dev/null") ne 0) {
+        record_info("Guest $self->{guest_name} unattended installation file hosted on local host can not be reached", "Mark guest installation as FAILED. The unattended installation file url is $self->{guest_installation_automation_file}", result => 'softfail');
```

Comment:
I still recommend to only use "softfail" in combination with ticket references

Reply:
No, we don't have or are going to create any ticket to solve the failure. It is a failure, we mark it softfail instead of fail is to continue the test. The test should not abort at here as we'd like the overall result of all the checks. Take an example, there are 7 checkpoints, the test is not supposed to fail at the 2nd one, we need checking them all then give a overaw result, marking it softfail to identify which checkpoints failed.